### DISCUSSION TITLE
Dynamic Lambda Cache Possible?

### DISCUSSION THREAD

**Discussion by jericopulvera** [vercel/vercel#4092] [General] [answered]
https://zeit.co/docs/v2/network/caching#serverless-functions-(lambdas)

I'm trying to create a functionality where each user would have a subdomain. 

The problem is when a subdomain is visited lambda would be cached and another subdomain would show the first one that's been cached. I would like it to be cached dynamically based on a given identifier.

Here's a sample code.
```
export async function getServerSideProps({ req, res }) {
    const siteIdentifier = req.headers.host.split('.')[0];
  
    if (process.env.NODE_ENV === 'development') siteIdentifier = 'local';
  
    res.setHeader('Cache-Control', `max-age=30,s-max-age=30,s-identifier${siteIdentifier}`);
  
    try {
      const { data: user } = await serverClient.query(
        q.Get(q.Match(q.Index('users_by_username'), siteIdentifier))
      );
  
      return { props: { user } };
    } catch (error) {
      return { props: { errorCode: 404 } };
    }
}
```

**Comment by coetry** [comment, marked as answer]
Hi @jericopulvera! We don't currently support this, but is a reasonable use case that has been brought up before and we are discussing implementation details. We'd love to make this possible, but need to consider the best approach.
